src/config_file.c

```diff
@@ -157,6 +158,141 @@ gboolean parse_bundle_formats(guint *mask, const gchar *config, GError **error)
+			continue;
+
+		logger_name = *group + strlen(RAUC_LOG_EVENT_CONF_PREFIX ".");
+
+		logger = g_new0(REventLogger, 1);
+		logger->name = g_strdup(logger_name);
+
+		logger->filename = key_file_consume_string(key_file, *group, "filename", NULL);
+		/* relative paths are resolved relative to the data-directory */
+		if (logger->filename && !g_path_is_absolute(logger->filename)) {
+			gchar *abspath;
+			if (!config->data_directory) {
+				g_set_error_literal(
+						error,
+						G_KEY_FILE_ERROR,
+						G_KEY_FILE_ERROR_INVALID_VALUE,
+						"Relative filename requires data-directory to be set");
+				return FALSE;
+			}
+
+			abspath = g_build_filename(config->data_directory, logger->filename, NULL);
+			g_free(logger->filename);
+			logger->filename = abspath;
+		}
+
+		log_format = key_file_consume_string(key_file, *group, "format", &ierror);
+		if (g_error_matches(ierror, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND)) {
+			log_format = g_strdup("readable");
+			g_clear_error(&ierror);
+		} else if (ierror) {
+			g_propagate_error(error, ierror);
+			return FALSE;
+		}
+
+		if (g_strcmp0(log_format, "readable") == 0) {
+			logger->format = R_EVENT_LOGFMT_READABLE;
+		} else if (g_strcmp0(log_format, "short") == 0) {
+			logger->format = R_EVENT_LOGFMT_READABLE_SHORT;
+		} else if (g_strcmp0(log_format, "json") == 0) {
+#if ENABLE_JSON
+			logger->format = R_EVENT_LOGFMT_JSON;
+#else
+			g_set_error(
+					error,
+					G_KEY_FILE_ERROR,
+					G_KEY_FILE_ERROR_INVALID_VALUE,
+					"Invalid log format %s. RAUC is compiled without JSON support", log_format);
+			return FALSE;
+#endif
+		} else if (g_strcmp0(log_format, "json-pretty") == 0) {
+#if ENABLE_JSON
+			logger->format = R_EVENT_LOGFMT_JSON_PRETTY;
+#else
+			g_set_error(
+					error,
+					G_KEY_FILE_ERROR,
+					G_KEY_FILE_ERROR_INVALID_VALUE,
+					"Invalid log format %s. RAUC is compiled without JSON support", log_format);
+			return FALSE;
+#endif
+		} else {
+			g_set_error(
+					error,
+					G_KEY_FILE_ERROR,
+					G_KEY_FILE_ERROR_INVALID_VALUE,
+					"Unknown log format '%s'", log_format);
+			return FALSE;
+		}
+
+		logger->events = g_key_file_get_string_list(key_file, *group, "events", &entries, NULL);
+		for (gsize j = 0; j < entries; j++) {
+			if (!r_event_log_is_supported_type(logger->events[j])) {
+				g_set_error(
+						error,
+						G_KEY_FILE_ERROR,
+						G_KEY_FILE_ERROR_INVALID_VALUE,
+						"Unsupported event type '%s'", logger->events[j]);
+				return FALSE;
+			}
+		}
+		g_key_file_remove_key(key_file, *group, "events", NULL);
+
+		logger->maxsize = key_file_consume_binary_suffixed_string(key_file, *group, "max-size", &ierror);
+		if (g_error_matches(ierror, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND)) {
+			g_clear_error(&ierror);
+		} else if (ierror) {
+			g_propagate_error(error, ierror);
+			return FALSE;
+		}
+
+		logger->maxfiles = key_file_consume_integer(key_file, *group, "max-files", &ierror);
```

Comment:
We should probably check that the provided value is in a sensible range (i.e. 1..1000).

Reply:
This still applies to the current version.

Reply:
We've already ensured "> 1" and I've now added a check for "< 1000"